components/browser/session/src/main/java/mozilla/components/browser/session/SessionManager.kt

```diff
@@ -122,7 +122,32 @@ class SessionManager(
      * @param snapshot A [Snapshot] which may be produced by [createSnapshot].
      * @param updateSelection Whether the selected session should be updated from the restored snapshot.
      */
-    fun restore(snapshot: Snapshot, updateSelection: Boolean = true) = delegate.restore(snapshot, updateSelection)
+    fun restore(snapshot: Snapshot, updateSelection: Boolean = true) {
+        delegate.restore(snapshot, updateSelection)
+
+        val tabs = snapshot.sessions.mapNotNull { item ->
+            if (!item.session.isCustomTabSession()) {
+                item.session.toTabSessionState()
+            } else {
+                // A restored snapshot should not contain any custom tab so we should be able to safely ignore
+                // them here.
+                null
+            }
+        }
+
+        val selectedTabId = if (updateSelection && snapshot.selectedSessionIndex != NO_SELECTION) {
+            val index = snapshot.selectedSessionIndex
+            if (index in 0..snapshot.sessions.lastIndex) {
```

Comment:
nit: index <= lastIndex?

Reply:
Not sure about this comment. We also need to check >= 0 (at least no selection is -1).

Reply:
Hm not sure what I wanted to say there :).

>We also need to check >= 0 (at least no selection is -1).
Yeah, makes sense.